src/main/java/net/openhft/chronicle/wire/DynamicEnum.java

```diff
@@ -20,7 +20,7 @@ static <E extends DynamicEnum> void updateEnum(E e) {
         E nums = cache.valueOf(e.name());
         List<FieldInfo> fieldInfos = e.$fieldInfos();
         for (FieldInfo fieldInfo : fieldInfos) {
-            fieldInfo.set(nums, fieldInfo.get(e));
+            fieldInfo.copy(e, nums);
```

Comment:
This assumed all fields were objects, which caused segfaults when we moved to unsafe